flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/dynamic/source/enumerator/DynamicKafkaSourceEnumeratorTest.java

```diff
@@ -464,6 +466,87 @@ public void testSnapshotState() throws Throwable {
         }
     }
 
+    @Test
+    public void testSnapshotStateMigration() throws Throwable {
```

Comment:
nit: 
```suggestion
    public void testEnumeratorStateDoesNotContainStaleTopicPartitions() throws Throwable {
```

or something similar

Reply:
Sounds good. Changed